chapter8/prompt-auto-optimization/coding_agent.py

```
@@ -60,21 +60,23 @@ def _apply_one(content: str, old_str: str, new_str: str) -> tuple[str, str | Non
 
 def _apply_edits_from_args(working: str, args: dict) -> tuple[str, int, list, list]:
-    """Apply apply_edits tool args to working text. JSON null edits like omit ([])."""
+    """Apply edits; null edits → []; skip non-dict entries with an error."""
     edits = args.get("edits")
     if edits is None:
         edits = []
     errors = []
     applied = 0
     for e in edits:
+        if not isinstance(e, dict):
+            errors.append(f"edit 必须是对象，收到 {type(e).__name__}")
```

Ruff (0.15.21)

[warning] String contains ambiguous `，` (FULLWIDTH COMMA). Did you mean `,` (COMMA)?

(RUF001)